ast/compile.go

```diff
@@ -1441,6 +1465,9 @@ func (c *Compiler) resolveAllRefs() {
 		for id, module := range parsed {
 			c.Modules[id] = module.Copy()
 			c.sorted = append(c.sorted, id)
+			if c.keepModules {
```

Comment:
💭 we could have used `if c.modules != nil` here, too, but it really doesn't matter, i think.